### PR TITLE
Add a way to publish finders with email signup

### PR DESCRIPTION
**Pull request by steventux**
https://trello.com/c/2BGHlD7M/13-specialist-publisher-publishing-finders

We need to publish a specialised finder which isn't tied to a particular document type much like advanced search. This finder needs to support email signup with multiple facets.

TODO:

- Add proposed facets
- Add/export signup links if necessary

screenshot from 2018-11-19 17-58-46

cc @thomasleese

### CODE REVIEW

File: config/find-eu-exit-guidance-business.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  - facet_id: sector
+    facet_name: Sector
+    facet_choices:
+    - key: aerospace
+      radio_button_name: Aerospace
+      topic_name: aerospace
+      prechecked: false
+    - key: agriculture-environment-and-natural-resources
+      radio_button_name: Agriculture, environment and natural resources
+      topic_name: agriculture-environment-and-natural-resources
```
Comment (thomasleese) [inline config/find-eu-exit-guidance-business.yml]: Topic name should be human readable. It's currently used to generate the title for the email subscription.

File: lib/publishing_api_finder_publisher.rb [hunk 2]
```diff
@@ -23,41 +23,105 @@ def publishing_api
...
+  end
+
+  def details
+    schema.fetch("details", {}).merge(
+      "subscription_list_title_prefix" => schema.fetch("subscription_list_title_prefix", {}),
+      "email_filter_facets" => schema.fetch("email_filter_facets", []),
+    ).except("document_noun", "facets", "reject")
+  end
+
+  def routes
```
Comment (thomasleese) [inline lib/publishing_api_finder_publisher.rb]: I may have missed it, but I don't think this method is being used.

File: config/find-eu-exit-guidance-business.yml [hunk 3]
```diff
@@ -12,7 +12,7 @@ rendering_app: finder-frontend
 details:
   beta: false
   document_noun: publication
-  subscription_list_title_prefix: Find EU Exit guidance for your business
+  subscription_list_title_prefix: 'Find EU Exit guidance for your business '
```
Comment (barrucadu) [inline config/find-eu-exit-guidance-business.yml:15]: The need for the trailing space makes me sad

File: config/find-eu-exit-guidance-business.yml [hunk 4]
```diff
@@ -0,0 +1,566 @@
+---
+base_path: "/find-eu-exit-guidance-business"
+content_id: 42ce66de-04f3-4192-bf31-8394538e0734
+signup_content_id: 2818d67a-029a-4899-a438-a543d5c6a20d
+title: Find EU Exit guidance for your business
+description: ''
+name: 'Find EU Exit guidance for your business'
+locale: en
+public_updated_at: '2018-11-23T09:00:00.000+00:00'
+publishing_app: specialist-publisher
```
Comment (barrucadu) [inline config/find-eu-exit-guidance-business.yml]: Can this be `rummager`?

File: config/find-eu-exit-guidance-business.yml [hunk 5]
```diff
@@ -0,0 +1,566 @@
...
+rendering_app: finder-frontend
+details:
+  beta: false
+  document_noun: publication
+  subscription_list_title_prefix: 'Find EU Exit guidance for your business '
+  reject:
+    sector: "_MISSING"
```
Comment (barrucadu) [inline config/find-eu-exit-guidance-business.yml]: Have we been able to test this base filter yet?

Comment (barrucadu) [inline config/find-eu-exit-guidance-business.yml]: Should be `sector_business_area`

Also, we should check that this works before merging.

**Review by barrucadu — CHANGES_REQUESTED** [review]

File: config/find-eu-exit-guidance-business.yml [hunk 6]
```diff
@@ -0,0 +1,566 @@
...
+    facet_choices:
+    - key: horizon-2020
+      radio_button_name: Horizon 2020
+      topic_name: Horizon 2020
+      prechecked: false
+    - key: cosme
+      radio_button_name: COSME
+      topic_name: COSME
+      prechecked: false
+    - key: european-investment-bank-eib-
```
Comment (barrucadu) [inline config/find-eu-exit-guidance-business.yml]: Trailing hyphens?

File: config/find-eu-exit-guidance-business.yml [hunk 7]
```diff
@@ -0,0 +1,566 @@
...
+      value: dont_know
+    display_as_result_metadata: true
+    filterable: true
+    key: employ_eu_citizens
+    name: Employ EU citizens
+    preposition: businesses that
+    type: text
+  - allowed_values:
+    - label: Do business in the EU
+      value: do_business_in_the_eu
```
Comment (barrucadu) [inline config/find-eu-exit-guidance-business.yml]: Some values use snake_case others use kebab-case

**Comment by thomasleese** [comment]
Tested an earlier version of this with the new finder-frontend multi facet email sign up which worked. Will test again when this is ready to be merged.

File: config/find-eu-exit-guidance-business.yml (hunk 3 above)
Comment (thomasleese) [reply config/find-eu-exit-guidance-business.yml:15]: Unfortunately precedent has already been set here as the current finder signups (albeit only single facet ones) include a trailing space on this value.

File: config/find-eu-exit-guidance-business.yml [hunk 8]
```diff
@@ -0,0 +1,566 @@
...
+description: ''
+name: 'Find EU Exit guidance for your business'
+locale: en
+public_updated_at: '2018-11-23T09:00:00.000+00:00'
+publishing_app: rummager
+rendering_app: finder-frontend
+details:
+  beta: false
+  document_noun: publication
+  subscription_list_title_prefix: Find EU Exit guidance for your business
```
Comment (barrucadu) [inline config/find-eu-exit-guidance-business.yml]: This seems to have lost its trailing space, it was `'Find EU Exit guidance for your business '`

Comment (steventux) [reply config/find-eu-exit-guidance-business.yml]: my bad rebase - reinstated.

**Review by barrucadu — APPROVED** [review]
I think we should test this after #1324 and #1325 have been merged and rummager reindexed.

**Pull request merged**